File: app/src/main/java/es/etologic/mahjongscoring2/data/local_data_source/local/daos/CombinationsDao.java
package es.etologic.mahjongscoring2.data.local_data_source.local.daos;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

import es.etologic.mahjongscoring2.domain.model.Combination;
import io.reactivex.Single;

@Dao
public interface CombinationsDao {

    @Insert
    void bulkInsert(List<Combination> combinations);

    @Query("SELECT * FROM Combinations " +
                   "WHERE combinationName LIKE :filter " +
                   "OR combinationDescription LIKE :filter " +
                   "ORDER BY combinationPoints, combinationName ASC")
    Single<List<Combination>> getFiltered(String filter);

    @Query("SELECT * FROM Combinations")
    Single<List<Combination>> getAll();
}

File: app/src/main/java/es/etologic/mahjongscoring2/data/local_data_source/local/daos/CombinationsDao.kt
package es.etologic.mahjongscoring2.data.local_data_source.local.daos

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import es.etologic.mahjongscoring2.domain.model.Combination
import io.reactivex.Single

@Dao
interface CombinationsDao {
    
    @Query("SELECT * FROM Combinations")
    fun getAll(): Single<List<Combination>>
    
    @Insert
    fun bulkInsert(combinations: List<Combination>)
    
    @Query(
        "SELECT * FROM Combinations " +
            "WHERE combinationName LIKE :filter " +
            "OR combinationDescription LIKE :filter " +
            "ORDER BY combinationPoints, combinationName ASC"
    )
    fun getFiltered(filter: String): Single<List<Combination>>
}
